rama-ua/src/emulate/service.rs

```diff
@@ -789,9 +789,12 @@ fn compute_sec_fetch_site_value(
                     let referer_protocol =
                         Protocol::maybe_from_uri_scheme_str_and_method(uri.scheme(), method);
 
-                    let default_port = uri
-                        .port_u16()
-                        .or_else(|| referer_protocol.as_ref().map(|p| p.default_port()));
+                    let default_port = uri.port_u16().or_else(|| {
+                        referer_protocol
+                            .as_ref()
+                            .map(|p| p.default_port())
```

Comment:
```suggestion
                            .and_then(|p| p.default_port())
```

^ so you do not need to flatten